locator/network_topology_strategy.cc

```diff
@@ -340,12 +340,16 @@ future<size_t> network_topology_strategy::calculate_min_tablet_count(schema_ptr
     if (tablet_hints.expected_data_size_in_gb) {
         tablet_count = std::max<size_t>(tablet_count, (tablet_hints.expected_data_size_in_gb.value() << 30) / target_tablet_size);
     }
-    if (tablet_hints.min_per_shard_tablet_count) {
-        tablet_count = std::max<size_t>(tablet_count, calculate_initial_tablets_from_topology(*s, tm, _dc_rep_factor, tablet_hints.min_per_shard_tablet_count.value()));
-    }
-    if (tablet_count == 0) {
-        tablet_count = co_await calculate_initial_tablets_from_topology(*s, tm, _dc_rep_factor) * initial_scale.value_or(1);
+
+    // If min_tablet_count is set, initial_scale should not be effective.
+    // This is for compatibility with the old "initial" tablet count option which suppresses automatic scaling.
+    auto min_per_shard_tablet_count = tablet_hints.min_per_shard_tablet_count.value_or(
+            tablet_hints.min_tablet_count ? 0 : initial_scale.value_or(1));
+    if (min_per_shard_tablet_count) {
```

Comment:
Don't we need to call `calculate_initial_tablets_from_topology` if `tablet_count == 0`, even if the min per-shard count is zero?

Reply:
hmm, that depends on whether `min_per_shard_tablet_count == 0` means that the option is unset (then your suggestion applies), or it means that we want to force no minimum, and go by avg tablet size alone. I assumed the latter.

Reply:
If someone wants to unset the option, he can remove it from tablet options.

Reply:
let's look at this again after rebase.

Reply:
My concern was about the default case for new tables where no option helped determine the tablet_count, so we defer to `calculate_initial_tablets_from_topology`.
The average tablet size for new tables would be 0.

Reply:
If there are no tablet options, then `min_per_shard_tablet_count` will pick up the default `initial_scale` option.

Reply:
ok, got it.